sklearn/model_selection/_search.py

```diff
@@ -1086,36 +1086,33 @@ def _store(key_name, array, weights=None, splits=False, rank=False):
         for key, param_result in param_results.items():
             param_list = list(param_result.values())
             try:
-                with warnings.catch_warnings():
-                    warnings.filterwarnings(
-                        "ignore",
-                        message="in the future the `.dtype` attribute",
-                        category=DeprecationWarning,
-                    )
-                    # Warning raised by NumPy 1.20+
-                    arr_dtype = np.result_type(*param_list)
+                arr = np.array(param_list)
             except (TypeError, ValueError):
```

Comment:
I _think_ the `TypeError` can now be removed it was specific to `np.result_type`. At least the tests seem to pass without it ...